pdns/syncres.hh

```diff
@@ -114,6 +114,38 @@ public:
     DNSName d_name;
     bool d_rdForward{false};
 
+    bool operator==(const AuthDomain& rhs) const
+    {
+      return
+        d_records == rhs.d_records &&
+        d_servers == rhs.d_servers &&
+        d_name == rhs.d_name &&
+        d_rdForward == rhs.d_rdForward;
+    }
+
+    [[nodiscard]] std::string toString(const std::string& indent = "", const std::string& indentLevel = "  ") const
```

Comment:
I think this is also not a generic toString(), more of a debug aid, so a better fitting name is needed.